Add support for @unpack, @pack! for file-interface (+Test,Documentation) (#455)
JuliaIO/JLD2.jl: README.md

end
```

> When additionally loading the [UnPack.jl](https://github.com/mauro3/UnPack.jl) package, the there defined `@unpack` and `@pack!` macros can be used to quickly save and load date from  the file-like interface. Example:

Reviewer: ```suggestion
When additionally loading the [UnPack.jl](https://github.com/mauro3/UnPack.jl) package, its `@unpack` and `@pack!` macros can be used to quickly save and load data from the file-like interface. Example:
```

Author: implemented in the new commit

@assert load("example.jld2", "mygroup/mystuff") == 42
```

The group `file["mygroup"]` can be accessed with the same file-like interface as the "full" struct. When using [UnPack.jl](https://github.com/mauro3/UnPack.jl), the macros `@unpack` and `@pack!` work without modifications.

> 

Reviewer: I think I'd prefer to have a separate section in the docs / readme titled something like `UnPack.jl API` 
rather than having it interleaved with the rest.

Personally, I haven't used these macros even once since the regular destructuring syntax was introduced.
(which can't really use here since it uses `getproperty`)

Author: Alright, good idea. Your suggestion is implemented in the new commit :+1:
